pkg/cloud/baremetal/actuators/machine/actuator.go

```diff
@@ -980,14 +1073,45 @@ func (a *Actuator) remediateIfNeeded(ctx context.Context, machine *machinev1beta
 			return a.deleteNode(ctx, node)
 		}
 
-		// node is deleted, we can power on the host
+		//we need this annotation to differentiate between unhealthy machine that
+		//needs remediation, and an unhealthy machine that just got remediated
+		return a.addPoweredOffForRemediationAnnotation(ctx, machine)
+	}
+
+	// here we know that host has been powered off and node has been deleted
+	if hasPowerOffRequestAnnotation(baremetalhost) {
+		// we can now power host back on
 		log.Printf("Requesting Host %s power on for Machine %s",
 			baremetalhost.Name, machine.Name)
-		return a.requestPowerOn(ctx, baremetalhost)
+		return a.requestPowerOn(ctx, machine, baremetalhost)
 	}
 
+	node, err := a.getNodeByMachine(ctx, machine)
+
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			log.Printf("Failed to get Node from Machine %s: %s", machine.Name, err.Error())
+			return err
+		}
+	}
 	//node is still not running, so we requeue
 	if node == nil {
+		if canReprovision(machine, baremetalhost) {
```

Comment:
If we can't reprovision, shouldn't we delete our annotations so we can try turning it off and on again?

Reply:
It will most probably only cause an infinite remediation loop. So I think this is the best approach at the moment. It can be improved in the future, with some logic in `MachineHealtCheck` to avoid such loops.